cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,472 @@
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :type aws_update_tag: int
+    :param aws_update_tag: Timestamp tag for tracking data freshness
+    :rtype: None
+    """
+    # Use MatchLink to create relationships between existing nodes
+    matchlink_schema = AssumedRoleMatchLink()
+
+    load_matchlinks(
+        neo4j_session,
+        matchlink_schema,
+        aggregated_role_assumptions,
+        lastupdated=aws_update_tag,
+        _sub_resource_label="AWSAccount",
+        _sub_resource_id=current_aws_account_id,
+    )
+
+    logger.info(
+        f"Successfully loaded {len(aggregated_role_assumptions)} role assumption relationships"
+    )
+
+
+def _aggregate_role_assumptions(
+    role_assumptions: List[Dict[str, Any]],
+) -> List[Dict[str, Any]]:
+    """
+    Aggregate role assumption events by (source_arn, destination_arn) pairs.
+
+    Combines multiple assumption events between the same source and destination
+    into a single record.
+
+    :param role_assumptions: List of role assumption events
+    :return: List of aggregated role assumption records
+    """
+    aggregated: Dict[tuple, Dict[str, Any]] = {}
+
+    for assumption in role_assumptions:
+        source_arn = assumption.get("SourcePrincipal")
+        dest_arn = assumption.get("DestinationPrincipal")
+        event_time = assumption.get("EventTime")
+
+        missing_fields = []
+        if not source_arn:
+            missing_fields.append("SourcePrincipal")
+        if not dest_arn:
+            missing_fields.append("DestinationPrincipal")
+        if not event_time:
+            missing_fields.append("EventTime")
+
+        if missing_fields:
+            logger.warning(
+                f"Skipping incomplete assumption record missing: {', '.join(missing_fields)}"
+            )
+            continue
+
+        # MyPy check is forcing this assert check because it thinks source_arn could be None. Could do a more explicit check above if this is a problem.
+        assert source_arn is not None
+        # Convert STS assumed-role ARNs to IAM role ARNs for source matching
+        normalized_source_arn = _convert_assumed_role_arn_to_role_arn(source_arn)
+
+        # Create aggregation key
+        key = (normalized_source_arn, dest_arn)
+
+        if key in aggregated:
+            # Update aggregated values
+            agg_data = aggregated[key]
+            agg_data["times_used"] += 1
+
+            # Update temporal bounds
+            if event_time < agg_data["first_seen"]:
+                agg_data["first_seen"] = event_time
+            if event_time > agg_data["last_used"]:
+                agg_data["last_used"] = event_time
+        else:
+            aggregated[key] = {
+                "source_principal_arn": normalized_source_arn,
+                "destination_principal_arn": dest_arn,
+                "times_used": 1,
+                "first_seen": event_time,
```

Comment:
This should be named something like "first seen in time window' because it could be confused with the `firstseen` field. We don't know when this field was first seen, we only know when in the given time window it was seen.